File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 7/14/2026, 2:04:41 PM
class Solution {
    public boolean isPalindrome(String s) {
        if(s.length()==0){
            return false;
        }String r=s.replaceAll("[^A-Za-z0-9]","");
        r=r.toLowerCase();
        int left=0,right=r.length()-1;
        while(left<right){
            if(r.charAt(left)!=r.charAt(right)){
                return false;
            }left++;
            right--;
        }return true;
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 7/14/2026, 2:04:47 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        s = re.sub(r'[^a-zA-Z0-9]', '', s)
        r=s[::-1]
        return r==s